platform/ethereum/compound/compound.go

```diff
@@ -0,0 +1,193 @@
+	for _, acc := range accounts {
+		ret1 := blockatlas.AccountLendingContracts{Address: acc.Address, Contracts: make([]blockatlas.LendingContract, 0)}
+		for _, t := range acc.Tokens {
+			asset := p.getSymbolByCSymbol(t.Symbol)
+			if len(req.Assets) > 0 && !sliceContains(asset, req.Assets) {
+				continue // not requested, skip
+			}
+			assetInfo := blockatlas.AssetInfo{Symbol: asset}
+			if ai, err := p.getAssetInfoForSymbol(asset); err == nil {
+				assetInfo = ai
+			}
+			ret1.Contracts = append(ret1.Contracts, blockatlas.LendingContract{
+				Asset:         assetInfo,
+				CurrentAmount: strconv.FormatFloat(asFloat(t.SupplyBalanceUnderlying.Value), 'f', 10, 64),
+			})
+		}
+		ret = append(ret, ret1)
+	}
+	return ret, nil
+}
+
+func getAssetInfo(t *CToken, includeMeta bool) blockatlas.AssetInfo {
+	ret := blockatlas.AssetInfo{
+		Symbol:         t.UnderlyingSymbol,
+		Description:    t.Name,
+		APY:            apyOfToken(t),
+		YieldPeriod:    0,
+		YieldFrequency: 15,
+		TotalSupply:    blockatlas.Amount(t.TotalSupply.Value),
+		MinimumAmount:  blockatlas.Amount("0"),
+		LockTime:       0,
+	}
+	if includeMeta {
+		ret.MetaInfo = blockatlas.AssetMetaInfo{
+			DefiInfo: &blockatlas.DefiAssetInfo{
+				AssetToken: blockatlas.DefiTokenInfo{
+					Symbol: t.UnderlyingSymbol,
+					Chain:  Chain,
+				},
+				TechnicalToken: blockatlas.DefiTokenInfo{
+					Symbol:          t.Symbol,
+					Chain:           Chain,
+					ContractAddress: t.TokenAddress,
+				},
+			},
+		}
+	}
+	return ret
+}
+
+func (p *Provider) getAssetInfoForSymbol(asset string) (blockatlas.AssetInfo, error) {
+	tokens, err := p.getTokensCached()
+	if err != nil {
+		return blockatlas.AssetInfo{}, err
+	}
+	token, ok := tokens[strings.ToUpper(asset)]
+	if !ok {
+		return blockatlas.AssetInfo{}, fmt.Errorf("Token not found %v", asset)
+	}
+	return getAssetInfo(&token, true), nil
+}
+
+func (p *Provider) getAssetInfos(includeMeta bool) ([]blockatlas.AssetInfo, error) {
+	res := make([]blockatlas.AssetInfo, 0)
+	tokens, err := p.getTokensCached()
+	if err != nil {
+		return res, err
+	}
+	for _, t := range tokens {
+		res = append(res, getAssetInfo(&t, includeMeta))
+	}
+	sort.Slice(res, func(i, j int) bool { return res[i].Symbol < res[j].Symbol })
+	return res, nil
+}
+
+func (p *Provider) getTokensCached() (map[string]CToken, error) {
+	tokens := make(map[string]CToken, 30)
+	res, err := p.client.GetCTokensCached(make([]string, 0), cacheValiditySec)
+	if err != nil {
+		return tokens, err
+	}
+	for _, t := range res.CToken {
+		tokens[strings.ToUpper(t.UnderlyingSymbol)] = t
+	}
+	return tokens, nil
+}
+
+func (p *Provider) getSymbolByCSymbol(symbol string) string {
+	tokens, err := p.getTokensCached()
+	if err == nil {
+		for s := range tokens {
+			if tokens[s].Symbol == symbol {
+				return s
+			}
+		}
+	}
+	return ""
+}
+
+func apyOfToken(token *CToken) float64 {
+	return 100.0 * asFloat(token.SupplyRate.Value)
+}
+
+func asFloat(value string) float64 {
```

Comment:
just use normal function to parse floats

Reply:
I prefer a helper func to be less verbose and because ParseFloat has 2 returns.